src/Microsoft.Diagnostics.Monitoring.StartupHook/StartupHook.cs

```diff
@@ -2,68 +2,47 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using Microsoft.Diagnostics.Monitoring.StartupHook;
-using Microsoft.Diagnostics.Monitoring.StartupHook.Exceptions;
-using Microsoft.Diagnostics.Monitoring.StartupHook.MonitorMessageDispatcher;
-using Microsoft.Diagnostics.Tools.Monitor;
-using Microsoft.Diagnostics.Tools.Monitor.HostingStartup;
-using Microsoft.Diagnostics.Tools.Monitor.Profiler;
-using Microsoft.Diagnostics.Tools.Monitor.StartupHook;
 using System;
-using System.IO;
 
 internal sealed class StartupHook
 {
-    private static CurrentAppDomainExceptionProcessor? s_exceptionProcessor;
-    private static AspNetHostingStartupLoader? s_hostingStartupLoader;
+    private static DiagnosticsBootstrapper? s_bootstrapper;
+    private static object s_lock = new object();
 
     public static void Initialize()
     {
-        try
```

Comment:
I take it there are no guards for this in the runtime itself?

Reply:
Correct. The runtime will only load the assembly once but it will call the `Initialize` method each time.